### PR TITLE
Refactor code and add policy/backend to swagger

### PR DESCRIPTION
**Pull request by RupengLiu**
Refactor some codes and add policy setting and backend entity to swagger import

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -2,7 +2,7 @@
     "name": "vscode-apimanagement",
     "displayName": "Azure API Management",
     "description": "An Azure API Management extension for Visual Studio Code.",
-    "version": "0.1.3-alpha",
+    "version": "0.1.4",
```
Comment (anganti) [inline package.json]: leave it to 3-alpha

File: src/azure/apim/policyComponents.ts [hunk 2]
```diff
@@ -0,0 +1,56 @@
+/*---------------------------------------------------------------------------------------------
```
Comment (anganti) [inline src/azure/apim/policyComponents.ts]: rename file as policyHelper.cs

File: src/azure/webApp/FunctionAppService.ts [hunk 3]
```diff
@@ -48,8 +48,13 @@ export class FunctionAppService {
     // Update function app config
     public async updateSiteConfigAPIM(resourceGroup: string, serviceName: string, apiName: string): Promise<void> {
         const webAppConfig: IWebAppContract = await this.getFuncAppConfig();
```
Comment (anganti) [inline src/azure/webApp/FunctionAppService.ts]: getWebAppConfig()?

File: src/azure/webApp/FunctionAppService.ts [hunk 4]
```diff
@@ -48,8 +48,13 @@ export class FunctionAppService {
...
-        if (webAppConfig.properties.apiManagementConfig.id === undefined) {
-            webAppConfig.properties.apiManagementConfig.id = `/subscriptions/${this.subscriptionId}/resourceGroups/${resourceGroup}/providers/Microsoft.ApiManagement/service/${serviceName}/apis/${apiName}`;
+        const apiConfigId = `/subscriptions/${this.subscriptionId}/resourceGroups/${resourceGroup}/providers/Microsoft.ApiManagement/service/${serviceName}/apis/${apiName}`;
+        if (webAppConfig.properties.apiManagementConfig === null) {
+            webAppConfig.properties.apiManagementConfig = {
+                id: apiConfigId
+            };
+        } else if (webAppConfig.properties.apiManagementConfig.id) {
+            webAppConfig.properties.apiManagementConfig.id = apiConfigId;
         }
```
Comment (anganti) [inline src/azure/webApp/FunctionAppService.ts]: Not sure if we want to replace the existing value. can we leave if the value is already set.?

Comment (RupengLiu) [reply src/azure/webApp/FunctionAppService.ts]: yes, let's leave it unchanged

File: README.md [hunk 5]
```diff
@@ -60,6 +63,30 @@ Please note: only JSON format is supported currently.
 
 ![Extract](resources/extract.gif)
 
+## Import Function App to create a new API
+![importFuncApp](resources/importFunc.gif)
+
+## Import Function App to an existing API
+![importFuncAppToApi](resources/importFuncToApi2.gif)
+
+## Import Web App (with swagger object) to create a new API
```
Comment (anganti) [inline README.md]: dont need all three. Just the happy path where swagger exists and leave other to documentation.

**Pull request merged**